### PR TITLE
Feature/add user class

### PR DESCRIPTION
**Pull request by Alvaro2112**
I reworked the whole implementation of the Player class, this used to be a mix of a User and a game-specific player. Now there are two classes, Player and User, the User is the object related to the account and it created a new Player each time it joins a new Game, so a  User is forever but a Player is only for a game. It took a bit of time to rework every single test and every single class as I had to be careful not to break anything and also because it was a lot of code to go through.

### CODE REVIEW

**Review by gauthierboeshertz — COMMENTED** [review]
You should change the name of the classes, for example the PlayerTest.java should be named UserTest

File: app/src/androidTest/java/sdp/moneyrun/player/PlayerTest.java [hunk 1]
```diff
@@ -48,7 +50,7 @@ public void setAddressWithDBUpdateWorks(){
         ValueEventListener listener = new ValueEventListener() {
             @Override
             public void onDataChange(@NonNull DataSnapshot snapshot) {
-                Player p = snapshot.getValue(Player.class);
+                User p = snapshot.getValue(User.class);
                 //player.setAddress(p.getAddress());
```
Comment (Tesa27) [inline app/src/androidTest/java/sdp/moneyrun/player/PlayerTest.java:55]: I think that this comment can be erased

**Review by Tesa27 — APPROVED** [review]
Generally this looks good!

File: app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import com.google.firebase.database.ValueEventListener;
+
+import sdp.moneyrun.user.User;
+
+public class UserDatabaseProxy extends DatabaseProxy {
+
+    private final String TAG = UserDatabaseProxy.class.getSimpleName();
+
+    private final String DATABASE_PLAYER = "users";
+    private final String DATABASE_PLAYER_SCORE = "score";
```
Comment (ArnaudPoletto) [inline app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java:17]: These could be changed to DATABASE_USER_...

**Review by ArnaudPoletto — CHANGES_REQUESTED** [review]
Otherwise, the overall changes are good

**Review by ArnaudPoletto — APPROVED** [review]
All good

**Pull request merged**